Microsoft.Toolkit.Uwp.UI.Controls/MasterDetailsView/MasterDetailsView.cs

```diff
@@ -216,30 +204,38 @@ private void SetMasterHeaderVisibility()
             }
         }
 
+        private void UpdateView(bool animate)
+        {
+            var currentState = ViewState;
+            UpdateViewState();
+            SetBackButtonVisibility(currentState);
+            if (_stateGroup != null)
+            {
+                SetVisualState(_stateGroup.CurrentState, animate);
+            }
+        }
+
         /// <summary>
         /// Sets the back button visibility based on the current visual state and selected item
         /// </summary>
-        private void SetBackButtonVisibility(VisualState currentState)
+        private void SetBackButtonVisibility(MasterDetailsViewState previousState)
         {
-            UpdateViewState();
             if (DesignMode.DesignModeEnabled)
             {
                 return;
             }
 
             if (ViewState == MasterDetailsViewState.Details)
             {
-                SystemNavigationManager.GetForCurrentView().AppViewBackButtonVisibility =
-                    AppViewBackButtonVisibility.Visible;
+                var navigationManager = SystemNavigationManager.GetForCurrentView();
+                _previousBackButtonVisibility = navigationManager.AppViewBackButtonVisibility;
+
+                navigationManager.AppViewBackButtonVisibility = AppViewBackButtonVisibility.Visible;
             }
-            else
+            else if (previousState == MasterDetailsViewState.Details)
             {
                 // Make sure we show the back button if the stack can navigate back
```

Comment:
This comment does not accurately describe the code block anymore. Could be updated or removed. 

Personally I think some explanation can be helpful here; I don't fully understand why the backbutton visibility should be set to a previous state? I think it should only be visible if the current state is MasterDetailsViewState.Details. But maybe I'm missing something: can you explain?